src/Events/Editor/Full_Site/Controller.php

```diff
@@ -0,0 +1,405 @@
+				'wp-i18n',
+				'wp-element',
+				'wp-editor',
+				'tribe-common-gutenberg-data',
+				'tribe-common-gutenberg-utils',
+				'tribe-common-gutenberg-store',
+				'tribe-common-gutenberg-icons',
+				'tribe-common-gutenberg-hoc',
+				'tribe-common-gutenberg-elements',
+				'tribe-common-gutenberg-components',
+			],
+			'enqueue_block_editor_assets',
+			[
+				'in_footer'    => false,
+				'localize'     => [],
+				'conditionals' => [ $this, 'is_full_site_editor' ],
+				'priority'     => 106,
+			]
+		);
+	}
+
+	/**
+	 * Internal FSE function for asset conditional testing.
+	 *
+	 * @since 5.14.2
+	 *
+	 * @return boolean Whether The current theme supports full-site editing or not.
+	 */
+	public function is_full_site_editor() {
+		return tec_is_full_site_editor();
+	}
+
+	/**
+	 * Adds the filters required by the FSE components.
+	 *
+	 * @since 5.14.2
+	 * @since TBD Adding support for block templates.
+	 */
+	protected function add_filters() {
+		add_filter( 'get_block_templates', [ $this, 'filter_include_templates' ], 25, 3 );
+		add_filter( 'get_block_template', [ $this, 'filter_include_template_by_id' ], 10, 3 );
+		add_filter( 'tribe_get_option_tribeEventsTemplate', [ $this, 'filter_events_template_setting_option' ] );
+		add_filter( 'tribe_get_single_option', [ $this, 'filter_tribe_get_single_option' ], 10, 3 );
+		add_filter( 'tribe_settings_save_option_array', [ $this, 'filter_tribe_save_template_option' ], 10, 2 );
+		add_filter( 'archive_template_hierarchy', [ $this, 'filter_archive_template_hierarchy' ], 10, 1 );
+		add_filter( 'single_template_hierarchy', [
+			$this,
+			'filter_single_template_hierarchy'
+		], 10, 1 );
+	}
+
+	/**
+	 * Removes registered filters.
+	 *
+	 * @since TBD
+	 */
+	public function remove_filters() {
+		remove_filter( 'get_block_templates', [ $this, 'filter_include_templates' ], 25 );
+		remove_filter( 'get_block_template', [ $this, 'filter_include_template_by_id' ], 10 );
+		remove_filter( 'tribe_get_option_tribeEventsTemplate', [ $this, 'filter_events_template_setting_option' ] );
+		remove_filter( 'tribe_get_single_option', [ $this, 'filter_tribe_get_single_option' ], 10 );
+		remove_filter( 'tribe_settings_save_option_array', [ $this, 'filter_tribe_save_template_option' ], 10 );
+		remove_filter( 'archive_template_hierarchy', [ $this, 'filter_archive_template_hierarchy' ], 10 );
+		remove_filter( 'single_template_hierarchy', [
+			$this,
+			'filter_single_template_hierarchy'
+		], 10 );
+	}
+
+	/**
+	 * Redirect the post type template to our Events Archive slug, as that is what is used for lookup in the database.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $templates Templates in order of display hierarchy.
+	 *
+	 * @return array Adjusted file name that is parsed to match our block template.
+	 */
+	public function filter_archive_template_hierarchy( $templates ) {
+		if ( empty( $templates ) ) {
+			return $templates;
+		}
+		if ( ! is_array( $templates ) ) {
+			return $templates;
+		}
```

Comment:
Note that we typically put line breaks on both ends of a conditional - except before the first line in a function.